### ISSUE TITLE
Question: Would you be open to some CI/CD?

### ISSUE THREAD

**Issue by gcarreno** [exilon/QuickLib#137]
Hey @exilon,

I have this thing that I kinda hate having tags without a GitHub official release.
I'm aware that this can be a me thing and that not everyone is on board.

I'm the author of [`setup-lazarus`](https://github.com/gcarreno/setup-lazarus), and I did this to allow me to have proper support for `CI`/`CD` with Free Pascal. Sorry, have tried to bug the Embarcadero peeps to give me something to do the same with Delphi, but haven't had any luck due to the fact that the Community Edition does not ship with a command line compiler.

If you tell me what is the project that you use for Unit testing, I can produce a `PR` that will compile and run it, at least with Free Pascal's `lazbuild`.

I could then do the same with `QuickLogger`, once we've had this one settled.

Many thanks in advance for your attention.

Cheers,
Gus

**Comment by gcarreno** [comment]
Hey John(@jkour),

After I dropped this issue, I realised that your code is more akin to a _package_ than an application.
That left me scratching my head a bit, on how I would do the `CI`/`CD` of it.
I guess that if you have a `CLI` unit testing app, I could compile and run that and that's the `CI` part done.
Then, the `CD` part is kinda just zipping the code... There's no cross compiling for many platforms... And that is already done on every release by default. So yeah, scratching my head :smiley: 
I guess we could automate the release process. Using something like a `release_notes.md` file on the repo to feed to the automated release action.

I don't quite remember if you have an example application. If so, that could be the thing that we can use for the `CD` part. An example, readily compiled into the 3 major platforms would be nice, right. If that even makes sense, of course.

In conclusion, I think I jumped the gun a tad.
I'll let you read my suggestions above and when you decide what you want, we can tailor it to your needs.

> I couldn't really understand how you use github actions with CI/CD

Please elaborate on this. You kinda lost me here...

Cheers,
Gus

**Comment by jkour** [comment]
Hi @gcarreno ,

I see. That's why I got confused as well 😃

For the record, I am not the owner of the repo. I just asked out of curiosity.

@exilon is the owner

**Comment by gcarreno** [comment]
Hey John(@jkour),

> I see. That's why I got confused as well 😃

Yeah, my deepest apologies for the confusion. I guess I shot my mouth without engaging my brain first :rofl: 

>For the record, I am not the owner of the repo. I just asked out of curiosity.
>@exilon is the owner

Yeaps, gotcha :wink: 

Nonetheless, I can still add the `CI` part ( if a `CLI` unit test thing is done ).
And on the `CD` part, I can add an automated release upon a `git push --tags` that will read the release notes from a file, e.g. `release_notes.md`.

So, @exilon, please give me a poke if you're up for it.

Cheers,
Gus
